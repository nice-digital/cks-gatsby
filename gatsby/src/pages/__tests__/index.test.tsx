import React from "react";
import { render } from "@testing-library/react";

import IndexPage from "../index";

describe("index", () => {
	it("should match snapshot", () => {
		const { container } = render(
			<IndexPage data={{ allTopics: { nodes: [] } }} />
		);

		expect(container.firstChild).toMatchSnapshot();
	});

	it("should render h1 with correct text", () => {
		const { queryByText } = render(
			<IndexPage data={{ allTopics: { nodes: [] } }} />
		);

		expect(queryByText("Clinical Knowledge Summaries")?.tagName).toBe("H1");
	});
});
