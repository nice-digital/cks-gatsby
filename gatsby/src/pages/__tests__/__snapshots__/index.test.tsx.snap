// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should match snapshot 1`] = `
<main
  class="container"
>
  <div
    class="hero"
  >
    <div
      class="hero__container"
    >
      <nav
        aria-label="Breadcrumbs"
        role="navigation"
      >
        <ol
          class="breadcrumbs"
        >
          <li
            class="breadcrumbs__crumb"
          >
            <a
              href="https://www.nice.org.uk/"
            >
              NICE
            </a>
          </li>
          <li
            class="breadcrumbs__crumb"
          >
            <span>
              CKS
            </span>
          </li>
        </ol>
      </nav>
      <script
        type="application/ld+json"
      >
        {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "https://www.nice.org.uk/",
        "name": "NICE"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "name": "CKS"
      }
    }
  ]
}
      </script>
      <div
        class="hero__body"
      >
        <div
          class="hero__copy"
        >
          <h1
            class="hero__title"
          >
            Clinical Knowledge Summaries
          </h1>
          <p
            class="hero__intro"
          >
            Providing primary care practitioners with a readily accessible summary of the current evidence base and practical guidance on bestÂ practice
          </p>
          <div
            class="hero__actions"
          >
            <a
              class="btn btn--cta"
              href="/topics/"
            >
              Topics A to Z
            </a>
            <a
              class="btn"
              href="/specialities/"
            >
              Specialities
            </a>
          </div>
        </div>
        <div
          class="hero__extra"
        >
          <h2
            class="h4 mt--0-md"
          >
            Most viewed topics
          </h2>
          <ul
            class="list--unstyled list--loose"
          >
            <li>
              <a
                href="/topics/lyme-disease/"
              >
                Lyme disease
              </a>
            </li>
            <li>
              <a
                href="/topics/hypertension-not-diabetic/"
              >
                Hypertension - not diabetic
              </a>
            </li>
            <li>
              <a
                href="/topics/diabetes-type-2/"
              >
                Diabetes - type 2
              </a>
            </li>
            <li>
              <a
                href="/topics/gout/"
              >
                Gout
              </a>
            </li>
            <li>
              <a
                href="/topics/menopause/"
              >
                Menopause
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <h2>
    Topics
  </h2>
  <ul />
</main>
`;
